File: arrays/noble-integer.cpp
int Solution::solve(vector<int> &A) {
    sort(A.begin(), A.end());
    int n = A.size();
    for(int i=0; i<n; i++){
        while(i<n-1 && A[i]==A[i+1]) i++;
        if(A[i]==n-1-i) return 1;
    }
    return -1;
}

File: arrays/noble-integer.py
class Solution:
    def solve(self, A):
        A.sort()
        for i in range(len(A)):
            if i < len(A)-1 and A[i+1] == A[i]: continue
            if A[i] == len(A) - 1 - i: return 1

        return -1

